Make use of Poetry parametrized

teutoburg opened the pull request:
The current "solution" of dealing with repos that use Poetry and also those that don't using two separate branches is not ideal. As we recently saw, any changes to the workflow in general have to be applied to both branches, which is cumbersome. I tried to let the workflow automatically decide which one to use based on the custom repo property that I included in most of our repos some time ago and has a Poetry flag. However, these (rather new in GH) properties are not directly accessible from the workflow, and getting them via the GH API and some creative JSON juggling might be doable but feels super hacky. Rather than that, I decided the best approach would be to add an optional parameter as a Poetry flag. This way, we can use the same workflow for all repos and simply pass a parameter if Poetry should be used.

To allow this change to be possible without breaking anything, I see two options:

1. Merge this into the `poetry` branch and make the flag True by default, that way any repos currently using this on the poetry branch without any parameter will get the same behavior as before. Any non-poetry repos are unaffected, but can be slowly switched over to using the `poetry` branch with the flag set to False (which is somewhat confusing, "use poetry but actually don't", oh well).
2. Call this branch `main` and don't merge it anywhere (which would mean closing the PR), and leave the flag to default to False (or not, doesn't actually make a functional difference in this case, as no current workflows would get there without manually changing the branch). Once all non-poetry repos are switched over to using this version as described in 1., the new `main` branch can be set as the default branch, and the old `master` discontinued. Similarly, once all poetry-based repos are using the new `main` branch with the flag True, we can discontinue the current `poetry` branch.

Which option do you prefer @hugobuddel ?

teutoburg commented:
FYI, I'm currently leaving this as a draft, even though I asked for opinions, because the changes to `tests.yml` should also be applied to `updated_tests.yml` and `webtests.yml`, but I'll do that after the above question is settled...

hugobuddel commented:
Thanks for doing the work! Renaming to `main` seems more sensible to me. Does require perhaps more repositories to change, but is less surprising.

We could also have different workflows, like `tests-poetry.yaml`. That is better than having multiple branches w.r.t. updating e.g. the Pyhon version or the action versions. But the `poetry` flag is also fine with me.

teutoburg commented:
> [...] w.r.t. updating e.g. the Pyhon version or the action versions.

Indeed the python versions were a main motivation for doing this now 🙃 

I think with the flag solution we'll have even less places to update things.

teutoburg commented:
Since we decided to go with option 2. and call this branch `main` without merging it into the `poetry` branch, I'm closing this PR.

teutoburg commented:
> 2\. Similarly, once all poetry-based repos are using the new `main` branch with the flag True, we can discontinue the current `poetry` branch.

I now changed the workflow files in all remaining repos that used the `poetry` branch over to `main`, making the former obsolete. Since `main` is based on `poetry`, I now deleted the `poetry` branch.
